### PR TITLE
Sensor Menu turned to generic menu per source

### PR DESCRIPTION
**Pull request by giltsuker**
- modified sensor menu to be created and implemented via source list, instead of hardcoding freq source and temp source
- de-featured temp source to only use psutil as the data sampling (i.e removed rasperry pie specific code)

### CODE REVIEW

File: s_tui/ComplexBarGraphs.py [hunk 1]
```diff
@@ -195,7 +202,8 @@ def set_title(self, title):
         if len(title) == 0:
             return
 
-        list_w = urwid.SimpleFocusListWalker([urwid.Text(title, align="center")])
+        list_w = \
```
Comment (amanusk) [inline s_tui/ComplexBarGraphs.py]: Could you remote the `\` ?

File: s_tui/Sources/FreqSource.py [hunk 2]
```diff
@@ -60,11 +60,11 @@ def __init__(self, is_admin):
         self.is_admin = is_admin
         self.is_available = True
 
-        self.top_freq = 0
+        self.top_freq = -1
         self.turbo_freq = False
         self.last_freq = 0
-        self.last_freq_list = [0] * len(psutil.cpu_freq(True))  # TODO compatiabiolty issues?
-        self.top_freq_list = [0] * len(psutil.cpu_freq(True))  # TODO compatiabiolty issues?
+        self.last_freq_list = [0] * len(psutil.cpu_freq(True))
```
Comment (amanusk) [inline s_tui/Sources/FreqSource.py]: This can fail if psutil.cpu_freq() is not implemented

File: s_tui/Sources/TemperatureSource.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 from __future__ import absolute_import
 
 import psutil
-import os
+# import os
```
Comment (amanusk) [inline s_tui/Sources/TemperatureSource.py]: If this is not used, just remove it

File: s_tui/StuiBarGraphVector.py [hunk 4]
```diff
@@ -186,7 +196,11 @@ def update_displayed_graph_data(self):
 
             graph.set_data(bars, float(max(data_max)))
             y_label_size_max = max(y_label_size_max, graph.get_size()[0])
+        # try:
         s = self.get_label_scale(0, max(data_max), float(y_label_size_max))
+        # except ValueError:
```
Comment (amanusk) [inline s_tui/StuiBarGraphVector.py]: Just remove the code if it is not needed

File: s_tui/s_tui.py [hunk 5]
```diff
@@ -198,50 +205,63 @@ class GraphView(urwid.WidgetPlaceholder):
...
+        self.source_list.append(TempSource(self.controller.custom_temp,
+                                           self.controller.temp_thresh))
+        self.source_list.append(FreqSource(is_admin))
 
+        # construct the variouse menus during init phase
         self.stress_menu = StressMenu(self.on_menu_close)
         self.help_menu = HelpMenu(self.on_menu_close)
         self.about_menu = AboutMenu(self.on_menu_close)
-        self.sensors_menu = SensorsMenu(self.on_sensors_menu_close, self.temp_source, self.freq_source)
+        self.sensors_menu = \
```
Comment (amanusk) [inline s_tui/s_tui.py]: Better to just add a new line after the `,`

File: s_tui/s_tui.py [hunk 6]
```diff
@@ -263,14 +283,16 @@ def update_displayed_information(self):
         self.clock_view.set_text(('bold text', seconds_to_text(
             int(self.controller.stress_time))))
 
+    # pycharm warning suppression for unused vars in function
```
Comment (amanusk) [inline s_tui/s_tui.py]: Is this a PyCharm byproduct?

File: s_tui/ComplexBarGraphs.py (hunk 1 above)
Comment (giltsuker) [reply s_tui/ComplexBarGraphs.py]: fixed

File: s_tui/Sources/FreqSource.py (hunk 2 above)
Comment (giltsuker) [reply s_tui/Sources/FreqSource.py]: fixed

File: s_tui/Sources/TemperatureSource.py (hunk 3 above)
Comment (giltsuker) [reply s_tui/Sources/TemperatureSource.py]: fixed

File: s_tui/StuiBarGraphVector.py (hunk 4 above)
Comment (giltsuker) [reply s_tui/StuiBarGraphVector.py]: fixed

File: s_tui/s_tui.py (hunk 5 above)
Comment (giltsuker) [reply s_tui/s_tui.py]: fixed

File: s_tui/s_tui.py (hunk 6 above)
Comment (giltsuker) [reply s_tui/s_tui.py]: pycharm gives weak warnings for some stuff, i added some comments to suppress all of those.
this isn't the only instance of this, i can remove them if it bothers you

**Comment by giltsuker** [comment]
- added a different way to handle summery texts with multiple sensors
- the travis 3.3 seems to have some sort of build error that is unrelated to s-tui

**Pull request merged**